src/movies.js

```diff
@@ -1,16 +1,113 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function getAllDirectors(array) {
+    // array is the array of movies
+    // your code in here!
+    // return a value
+    let directorsArr = array.map((movie) => {
+        return movie.director
+    })
+    return directorsArr
+  }
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array) {
+
+let filterArray=array.filter ((movie)=>{
+    return movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+}) 
+        return filterArray.length
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies) {  
+
+    if (movies.length === 0) {
+        return 0
+    }
+
+    let total = movies.reduce ((acc,elem) => {  
+        return acc+elem.rate 
```

Comment:
the test that is not passing could be handled here. Sometimes a movie may not have a rate, so trying to add that value (undefined) to the acc will result on an issue. Creating a condition here would solve it!